cpp/src/arrow/dataset/partition_test.cc

```diff
@@ -959,6 +1025,16 @@ TEST_F(TestPartitioning, Range) {
                          less_equal(field_ref("z"), literal(3.0)))}));
 }
 
+TEST_F(TestPartitioning, RangePartitoningEquals) {
```

Comment:
Not sure this check is actually useful? `RangePartitioning` is just a test class...

Reply:
yeah that's true. I just added it. I will remove this one.